src/index.ts

```diff
@@ -290,3 +291,50 @@ function scenarioNotAuthorised(
     authenticated: false
   }
 }
+
+
+
+// ㊙️
+
+
+async function importClassifiedInfo(
+  classifiedParam: Maybe<string>
+): Promise<void> {
+  if (!classifiedParam) return
+
+  const ks = await keystore.get()
+  const classifiedInfo = JSON.parse(common.base64.urlDecode(classifiedParam))
+
+  // Extract session key and its iv
+  const iv = utils.base64ToArrBuf(classifiedInfo.iv)
+  const rawSessionKey = await ks.decrypt(classifiedInfo.sessionKey)
+  const sessionKey = await crypto.subtle.importKey(
+    "raw",
+    utils.base64ToArrBuf(rawSessionKey),
+    "AES-GCM",
+    false,
+    [ "encrypt", "decrypt" ]
+  )
+
+  // Decrypt secrets
+  const secrets: Record<string, { key: string, bareNameFilter: string }> =
+    JSON.parse(utils.arrBufToStr(await crypto.subtle.decrypt(
```

Comment:
and we have decryption helper functions in `webnative/keystore`

is there some reason not to use those?

Reply:
Yeah I didn't use that because it does an `importKey` and `decrypt` in one function, and here I need to do multiple decrypts with the same key. I only wanted to do one key import.

Reply:
ah got it got it. you can do this with `keystore.aes.encrypt` as well then to avoid multiple imports :ok_hand: 

it accepts either a key string or an imported key as a param